hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/db/RDBRWStoreIterator.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ *  with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.hadoop.hdds.utils.db;
+
+import java.io.IOException;
+import org.apache.hadoop.hdds.utils.db.managed.ManagedRocksIterator;
+
+/**
+ * RocksDB ReadWrite store iterator.
+ */
+public class RDBRWStoreIterator extends RDBStoreIterator {
+  private RWBatchOperation rwBatch;
+
+  public RDBRWStoreIterator(
+      ManagedRocksIterator iterator, RWBatchOperation rwBatch,
+      RDBTable table) throws IOException {
+    super();
+    initIterator(iterator, table);
```

Comment:
@errose28 
Can not use RDBStoreIterator constructor because it performs additional step of seekToFirst() and this Must happen after lock, else concurrent close can cause rocksDB crash as observed. So seekToFirst() is called in this constructor after lock.

Reply:
@errose28 Updated as per RefCount, the above change is reverted